src/Config/Settings.ts

```diff
@@ -33,8 +33,31 @@ export const TASK_FORMATS = {
 
 export type TASK_FORMATS = typeof TASK_FORMATS; // For convenience to make some typing easier
 
+export class GlobalFilter {
+    value: string;
+    length: number;
+
+    constructor(value: string) {
+        this.value = value;
+        this.length = value.length;
+    }
+
+    matches(searchIn: string): boolean {
+        return searchIn.includes(this.value);
+    }
+
+    removeFrom(aString: string): string {
+        return aString.replace(this.value, '').trim();
+    }
+}
+
+export const getGlobalFilter = (): GlobalFilter => {
+    const { globalFilter } = getSettings();
+    return globalFilter;
```

Comment:
Maybe I didn't explain it very well, but the idea behind the proposal in #1783 to create an abstraction for the global filter is to hide all the implementation details of the global filter inside the new class.

When we talk next, let's talk about how abstractions work, and what this code might look like if the settings were fully abstracted and hidden away.